frame/composable-traits/src/vamm.rs

```diff
@@ -11,18 +15,27 @@ use frame_support::pallet_prelude::DispatchError;
 pub trait Vamm {
 	/// The balance type for an account.
 	type Balance;
-
+	/// Signed fixed point number implementation
+	type Decimal: FixedPointNumber;
+	/// Configuration for creating and initializing a new vAMM instance. May be used in extrinsic
+	/// signatures
+	type VammConfig: FullCodec + MaxEncodedLen + TypeInfo + Debug + Clone + PartialEq;
 	/// The identifier type for each virtual automated market maker.
-	type VammId;
+	type VammId: FullCodec + MaxEncodedLen + TypeInfo;
 
 	/// Create a new virtual automated market maker.
 	///
 	/// ## Returns
 	/// The identifier of the newly created vamm.
-	fn create(config: VammConfig<Self::Balance>) -> Result<Self::VammId, DispatchError>;
+	fn create(config: Self::VammConfig) -> Result<Self::VammId, DispatchError>;
```

Comment:
@Cardosaum May I suggest changing this to take in a *reference* to a `VammConfig` instead? I see no particular reason why `create` should take ownership of `config`. It also causes the Clearing House to have to call `vamm_config.clone()` when passing this value to the function in question, which might have an unnecessary resource cost.

If don't have any objections, I can include this change in this PR already.